### DISCUSSION TITLE
Roadmap

### DISCUSSION THREAD

**Discussion by GeorgiaHCA** [alan-turing-institute/AutSPACEs#376] [General]
This issue will become the Roadmap for the AutSPACEs platform development. 

Creating this roadmap resolves (#33)[https://github.com/alan-turing-institute/AutSPACEs/issues/33]

The Roadmap will also be added to the repository as a markdown doc. 

## Current Milestones

1. OH Integration - complete
2. Build MVP - underway 
3. Make MVP functional with Open Humans - underway
4. Accessibility checks - upcoming
5. Deployment of MVP - upcoming
6. Platform second phase with more of the requirement doc features - later stage

**other milestones ideas:**
create MVP prototype based on spec doc
make MVP functional with open humans
Create pathways for different user types (e.g. moderators)
User testing (in multiple cycles)
Integrate basic features with Open Humans
Build moderation process
Interface design
Add additional features desired by community
Accessibility checks
Create functional platform (more advanced than MVP)
Deployment
Launch

**Comment by GeorgiaHCA** [comment]
The Turing Way has a chapter on Roadmapping: https://the-turing-way.netlify.app/collaboration/github/github-roadmapping.html

**Comment by sowmyaarajan** [comment]
@GeorgiaHCA thank you for assigning. I am looking into it 😄

**Comment by sowmyaarajan** [comment]
Ideas or Questions to the AutSPACEs Repo

- Creating a Miro board to get a more interactive answers regarding the development from the community.

- Making module blocks and labelling them as beginner, intermediate and advance.

- Is it okay to do User testing only on the Landing Page and the experiences page?

**Comment by sowmyaarajan** [comment]
> The Turing Way has a chapter on Roadmapping: https://the-turing-way.netlify.app/collaboration/github/github-roadmapping.html

@GeorgiaHCA also, do we need a Static roadmap or a dynamic one? I was thinking to use few roadmap tools online. What do you feel about that? 😄

**Comment by GeorgiaHCA** [comment]
Hiya, I think there's a lot of detail needed here, but some ideas for roadmap milestones relating to launching the platform (not necessarily ordered).

* create MVP prototype based on spec doc
* make MVP functional with open humans 
* Create pathways for different user types (e.g. moderators) 
* User testing (in multiple cycles)
* Integrate basic features with Open Humans
* Build moderation process
* Interface design 
* Add additional features desired by community 
* Accessibility checks
* Create functional platform (more advanced than MVP)
* Deployment
* Launch

**Comment by GeorgiaHCA** [comment]
Heya! Here's a rough sketch of what it seems like some of the main milestones are for the platform build, but obvs I don't have so much tech background @sowmya9507, @BrainonSilicon, @jhlink, @Islast - then the idea would be to fill in the details after: so chunck it up into labels, and then have small, manageable issues for separate tasks (and all these would link to a broad roadmap)

What's most helpful to you to create the roadmap @lottycoupat?

**Comment by lottycoupat** [comment]
@GeorgiaHCA This is really helpful to see! I think for me also would be the tasks that would be needed for each milestone and what the dependencies are for each. For example, what needs to be done before we can start the next task and what tasks could be done at the same time. 

The idea suggested by @sowmya9507 above is also really helpful in terms of labelling these as Beginner/Intermediate/Advanced, so that those who want to be involved can see what they can pick up...

Does that sound do-able?

**Comment by GeorgiaHCA** [comment]
I do love that idea and also Kirstie's suggestion of giving time estimates.

**Comment by GeorgiaHCA** [comment]
I had a go at that for the project management repo: https://github.com/alan-turing-institute/AutisticaCitizenScience/issues/419 - We can credit @SuziQpid for her work and ideas for this. 

I think there's likely to be some technical skills we could assume for working in the AutSPACEs repo so beginner is maybe a bit of a higher bar than in the other repo. It would need to be clear in some way, either by explaining directly or changing the issues slightly?

**Comment by lottycoupat** [comment]
Potentially if those labels have already been created, we don't change their definition but only have intermediate/advanced in this repo? Avoid beginner labels as a beginner with no technical skills wouldn't be able to pick up the dev work? I think that would make it clear and avoid confusion...? 

Or there is also a "tech-skills-req" label which we could use? If it's easier than intermediate?

**Comment by SuziQpid** [comment]
Dear All,

Perhaps we could change beginner label to ‘entry or basic  tech skills’ ( BTS)  for simpler tasks than intermediate.

I don’t qualify for even that category and l think Lotty is right. We need to play to our strengths, avoid confusion whilst always being inclusive of everyone who wants to make a contribution big or small. A tall task of course. 

Suzanne lwai

**Comment by lottycoupat** [comment]
@SuziQpid Thanks Suzanne! Potentially we change the tech-skills-req to "Basic Tech Skills" so that we still have the beginner one for other tasks? 

@GeorgiaHCA we can make this clear in a contributor skills doc?

**Comment by jhlink** [comment]
Jumping in because my life revolves around Github. :D 

Why not use the `Good First Task` label for beginner level tasks? 
The kind of tasks I expect/create with this label are adding some HTML elements, changing some text, or generally something that is commonly practiced/implemented and can be easily Google'd. If you can copy/pasta the code from somewhere else and it works here, it's a simple enough task, imo.

**Comment by lottycoupat** [comment]
So for platform development we would have Good first task --> Intermediate --> Advanced - if so, I like that!! @jhlink

**Comment by GeorgiaHCA** [comment]
Good first task is geared towards people who are contributing for the first time to GitHub and it is a label which functions across repositories so seems a good one to have!

**Comment by GeorgiaHCA** [comment]
I also like @sowmya9507's idea of having a flow chart (/decision tree?) to help people work out which issues they can do, although I expect that could be a lot of work!

**Comment by lottycoupat** [comment]
@GeorgiaHCA @sowmya9507 I am working on some contributing skills with involve setting out the different areas of the project, what skills they need for it and how to get involved. For the how to get involved, I have what labels they should look out for in the repo. Which sounds familiar. Perhaps once that's completed we can see the need to map out into a visualisation/decision tree and balance capacities and other work?

**Comment by GeorgiaHCA** [comment]
Yep, makes sense @lottycoupat :)

**Comment by SuziQpid** [comment]
Dear All,

A little triggered by the word Good attached to any task ! If you fail at it are you ‘bad ‘ at basic technology.?!  I recognise though that if that’s a commonly used term across Git hub ld have to bow to the majority of users.

I can’t visualise a decision tree though the concept sounds great. Do you attach your name to a branch of it that describes the type of work / features that are being discussed.  

Suzanne lwai

**Comment by GeorgiaHCA** [comment]
Thanks for joining the discussion @SuziQpid ! And for taking on GitHub!

**Comment by GeorgiaHCA** [comment]
A decision tree is a flow chart you use to understand what things are included when making a decision. Here's a simple example:

![image](https://user-images.githubusercontent.com/46889966/118015199-ee7caf80-b34b-11eb-8351-07325252c869.png)

**Comment by GeorgiaHCA** [comment]
I think "good first issue" is meant to say whether _the issue_ is good or bad for new people, not whether the person working on the issue is bad, but I can see how that could be unclear! Making everything clear is a really important part of the onboarding process so thank you for pointing that out.

**Comment by GeorgiaHCA** [comment]
@lottycoupat - tagging you as we made some changes to the dev milestones (see top comment on issue)

**Comment by Islast** [comment]
Linking to our milestones page: https://github.com/alan-turing-institute/AutSPACEs/milestones

> Heya! Here's a rough sketch of what it seems like some of the main milestones are for the platform build, but obvs I don't have so much tech background @sowmya9507, @BrainonSilicon, @jhlink, @Islast - then the idea would be to fill in the details after: so chunck it up into labels, and then have small, manageable issues for separate tasks (and all these would link to a broad roadmap)

I think a good roadmap has a brief description of what the projects goals are; a sequence of milestones that function as a kind of narrative of how we're going to reach those goals; and finally a note of any things that haven't been accounted for in any of our milestones. Once the milestones are laid out it's a lot easier to break them down into tasks.

I think we're in a good place with the milestones we have already. I really like that we're commenting next to the milestones to say what our status is on each of these too. If we clean it up a little bit I think it's ready to be added to the repo as a markdown file.

**Comment by GeorgiaHCA** [comment]
@sowmya9507 , @lottycoupat and @anoura12 - here's the roadmap we were working on. It needs to be a in a structured md which links to the milestones and issues

**Comment by gedankenstuecke** [comment]
I've converted this into a discussion for now as I'm not sure if this is still a place to discuss the roadmap or whether we're past this for now. @GeorgiaHCA feel free to delete this discussion if it's too outdated :)

**Comment by SuziQpid** [comment]
As lm only an occasional contributor to moderation and display interface and probably will help test end user functions l think this list is super useful. I'd echo Kirstie and Georgia with which tasks naturally progress / or liberate flow to others, which can we do side by side so intersects might show up/ necessary tweaks to improve functionality. Of course lm a beginner task person. Also do we have enough volunteer personnel to fulfill each level of task beginner, intermediate and advanced as we start 2023.The flow of tasks can't solely be driven by what they are, rather whos able to work on them. Is there a need to recruit more people. Kudos to the core team who are always. Happy holidays to everyone.
